pubsublite/pscompat/settings.go

```diff
@@ -164,6 +164,28 @@ type NackHandler func(*pubsub.Message) error
 // error and terminate.
 type ReceiveMessageTransformerFunc func(*pb.SequencedMessage, *pubsub.Message) error
 
+// ReassignmentHandlerFunc is called any time a new partition assignment is
+// received from the server. It will be called with both the previous and new
+// partition numbers as decided by the server.
+//
+// When this handler is called, partitions that are being assigned away are
```

Comment:
Do we offer any stronger guarantees? For example: can we say something like "when this is invoked, no more messages from the removed partitions will be delivered to the callback?" Can the message callback be invoked concurrently to this callback?

Reply:
I added "message deliveries may still be in flight" to the comments. For Go, we can't offer strong guarantees. The Java implementation does, but Python doesn't.

I spent some time trying to offer the strong guarantee for Go, but it requires placing a mutex around the invocation of the message callback [here](https://github.com/googleapis/google-cloud-go/blob/320f4970ccd396d6af69486862730cff1b4b3e28/pubsublite/internal/wire/subscriber.go#L120), as well as reading/writing a "shutdown" state. But the downside is that a message receiver blocking for too long can delay acking the new assignment, which I think is what the [Java implementation](https://github.com/googleapis/java-pubsublite/blob/00b6863fc2521919048f0d582fa963e98a13c915/google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/cloudpubsub/internal/AssigningSubscriber.java#L92) has.